### PR TITLE
Sonar cloud implementation

### PR DESCRIPTION
**Pull request by Ali-chakaroun**

### CODE REVIEW

File: analyzer-java/pom.xml [hunk 1]
```diff
@@ -17,6 +17,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
 
+
```
Comment (wandernauta) [inline analyzer-java/pom.xml:20]: Nit: Change seems unneeded.

File: analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/AnalysisVisitor.java [hunk 2]
```diff
@@ -239,11 +239,11 @@ public List<Description> visit(JavadocComment n, Analyzer arg) {
...
-    // This regex splits the sentence into 2 if it ends with a . is followed by empty space.
-    String[] sentences = CommentHelperMethods.extractSummary(n).split("\\.\\s+", 2);
-    // Give back the . that is removed by the split.
-    String summary = (sentences.length > 0) ? sentences[0]+"." : null;
-    String remarks = (sentences.length > 1) ? sentences[1] : null;
+    // This regex splits the sentence into 2 at the first dot(.) it finds.
+    String[] sentences = CommentHelperMethods.extractSummary(n).split("\\.", 2);
+    // Add a dot(.) at the end if it is missing.
+    String summary = (sentences.length > 0) ? sentences[0].strip().concat(".") : null;
+    String remarks = (sentences.length > 1) ? sentences[1].strip() : null;
```
Comment (wandernauta) [inline analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/AnalysisVisitor.java:246]: Can you rebase this PR on `main`, without the changes to comments? That way, we can merge the two things separately.

File: .github/workflows/build.yml
Comment (wandernauta) [inline .github/workflows/build.yml:1]: Suggest rename to `sonar.yml` or similar.

File: pom.xml [hunk 3]
```diff
@@ -7,7 +7,10 @@
   <artifactId>livingdocumentation-java</artifactId>
   <version>1-SNAPSHOT</version>
   <packaging>pom</packaging>
-
+  <properties>
+    <sonar.organization>isep-livingdocumentation</sonar.organization>
+    <sonar.host.url>https://sonarcloud.io</sonar.host.url>
+  </properties>
```
Comment (wandernauta) [inline pom.xml:13]: Would it make sense to move these to the YAML file you're adding above, as a `-D` property? They're only really needed in the Sonar build, I think.

**Review by wandernauta — COMMENTED** [review]
Neat! Sonarcloud looks like a really polished tool, and I agree with all of its findings in this case.

I am a bit surprised it says "Coverage 0.0%". Can we add coverage tracking here? Or is that a premium-only feature or something like that?

**Comment by Ali-chakaroun** [comment]
I believe the coverage will fix itself once it's part of main.

File: pom.xml (hunk 3 above)
Comment (Ali-chakaroun) [reply pom.xml:13]: im not sure since these are the steps SonarCloud provide to add to the pipeline. I will look into it.

**Comment by wandernauta** [comment]
> I believe the coverage will fix itself once it's part of main.

Based on the instructions here...

https://docs.sonarcloud.io/enriching/test-coverage/java-test-coverage/

...I would expect that additional changes to `pom.xml` would be needed for code coverage tracking to work. Did you track coverage when you used SonarCloud earlier? If so, did that work without adding JaCoCo?

**Comment by Ali-chakaroun** [comment]
ignore this branch its too broken and refuses to rebase properly.

**Pull request closed without merging**